File: FE/jewelry/src/service/ProductService.js
import { log } from "async";
import axios from "axios"

export const getProductById = async (id) => {
    try {
        const res = await axios.get(`http://localhost:8080/product/${id}`)
        return res;
    } catch (e) {
        console.log(e);
    }
}
export const getListSearch = async (page, nameProduct, choose, search) => {
    try {
        const res = await axios.get(`http://localhost:8080/product/page-list?page=${page}&size=4&nameProduct=${nameProduct}&choose=${choose}&search=${search}`)
        console.log("service", res);
        return res;
    } catch (e) {
        console.log(e);
    }
}
export const getListHeader = async (page, size, nameProduct, nameType, nameCategory, nameTrademark, sort, sortBy) => {
    const specialCharactersRegex = /[!@#$%^&*()_+{}\[\]:;<>,.?~\\/-`]/;
        if(specialCharactersRegex.test(nameProduct)){
            return {status:204};
        } else{
    try {
        const res = await axios.get(`http://localhost:8080/product/page-header?page=${page}&size=${size}&nameProduct=${nameProduct}&nameType=${nameType}&nameCategory=${nameCategory}&nameTrademark=${nameTrademark}&sort=${sort}&sortBy=${sortBy}`)
        return res;
    } catch (e) {
        console.log(e);
    }
}
}
export const getSearchType = async(page,nameType) => {
    try{
        const res = await axios.get(`http://localhost:8080/product/type-header?page=${page}&nameType=${nameType}`)
        return res;
    }catch(e){
        console.log(e);
    }
}
export const getSearchCategory = async(page, nameCategory) => {
    try{
        const res = await axios.get(`http://localhost:8080/product/category-header?page=${page}&nameCategory=${nameCategory}`)
        return res;
    }catch(e){
        console.log(e);
    }
}
export const getSearchTrademark = async(page,nameTrademark) => {
    try{
        const res = await axios.get(`http://localhost:8080/product/trademark-header?page=${page}&nameTrademark=${nameTrademark}`)
        return res;
    }catch(e){
        console.log(e);
    }
}

File: BE/Jewerly/src/main/java/com/example/jewerly/product/service/ProductService.java
package com.example.jewerly.product.service;

import com.example.jewerly.product.dto.IProductDto;
import com.example.jewerly.product.model.Category;
import com.example.jewerly.product.model.Trademark;
import com.example.jewerly.product.model.Type;
import com.example.jewerly.product.repository.ICategoryRepository;
import com.example.jewerly.product.repository.IProductRepository;
import com.example.jewerly.product.repository.ITrademarkRepository;
import com.example.jewerly.product.repository.ITypeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ProductService implements IProductService{
    @Autowired
    private IProductRepository productRepository;
    @Autowired
    private ITypeRepository typeRepository;
    @Autowired
    private ICategoryRepository categoryRepository;
    @Autowired
    private ITrademarkRepository trademarkRepository;
    @Override
    public IProductDto getProductById(Integer id) {
        return productRepository.findProductById(id);
    }

    @Override
    public List<IProductDto> getListHome() {
        return productRepository.getListNew();
    }

    @Override
    public List<IProductDto> getListBestSeller() {
        return productRepository.getListBestseller();
    }

    @Override
    public List<Category> categoryList() {
        return categoryRepository.categoryList();
    }

    @Override
    public List<Type> typeList() {
        return typeRepository.typeList();
    }

    @Override
    public List<Trademark> trademarkList() {
        return trademarkRepository.trademarkList();
    }

    @Override
    public Page<IProductDto> getPageList(String nameProduct, String nameType, String nameCategory, String nameTrademark, Pageable pageable) {
        return productRepository.getListSearch(nameProduct, nameType, nameCategory, nameTrademark, pageable);
    }

    @Override
    public Page<IProductDto>    getPageType(String nameType, Pageable pageable) {
        return productRepository.getListType(nameType, pageable);
    }

    @Override
    public Page<IProductDto> getPageCategory(String nameCategory, Pageable pageable) {
        return productRepository.getListCategory(nameCategory, pageable);
    }

    @Override
    public Page<IProductDto> getPageTrademark(String nameTrademark, Pageable pageable) {
        return productRepository.getListTrademark(nameTrademark, pageable);
    }
}
